.circleci/config.yml

```diff
@@ -1,165 +1,67 @@
-      - run: |
-          bundle check --path vendor/bundle || bundle install --path vendor/bundle
-          bundle clean
-      - save_cache:
-          key: <<parameters.cache_key>>-{{ arch }}-{{ .Branch }}-{{ checksum "<<parameters.checksumfile>>" }}
-          paths:
-            - vendor/bundle
-  rspec-unit:
-    parameters:
-      additional_args:
-        type: string
-        default: ""
-      glob:
-        type: string
-        default: ""
-    steps:
-      - run: mkdir ~/rspec
-      - run:
-          name: "Run rspec tests"
-          command: |
-            TESTFILES=$(circleci tests glob "spec/**/*_spec.rb" | circleci tests split --split-by=timings)
-            echo "Running: ${TESTFILES}"
-            bundle exec rspec --format progress --format RspecJunitFormatter -o ~/rspec/rspec.xml <<parameters.additional_args>> -- ${TESTFILES}
-          when: always
-      - store_test_results:
-          path: ~/rspec
-  bundle-audit:
-    parameters:
-      additional_args:
-        type: string
-        default: ""
-    steps:
-      - run: bundle exec bundle-audit update
-      - run: bundle exec bundle-audit check -v <<parameters.additional_args>>
-  rubocop:
-    steps:
-      - run: bundle exec rubocop -p
+ruby_3_0_defaults: &ruby_3_0_defaults
+  <<: *defaults
+  e:
+    name: ruby/ruby
+    ruby-version: '3.0'
 
-jobs:
-  bundle-audit:
-    executor: <<parameters.e>>
-    parameters:
-      e:
-        type: executor
-        default: "ruby_2_6"
-    steps:
-      - pre-setup
-      - bundle-install
-      - bundle-audit
-  rubocop:
-    executor: <<parameters.e>>
-    parameters:
-      e:
-        type: executor
-        default: "ruby_2_6"
-    steps:
-      - pre-setup
-      - bundle-install
-      - bundle-audit
-  rspec-unit:
-    executor: <<parameters.e>>
-    parameters:
-      e:
-        type: executor
-        default: "ruby_2_6"
-    steps:
-      - pre-setup
-      - bundle-install
-      - rspec-unit
+ruby_3_1_defaults: &ruby_3_1_defaults
+  <<: *defaults
+  e:
+    name: ruby/ruby
+    ruby-version: '3.1'
 
 workflows:
   version: 2
-  ruby_2_6:
-    jobs:
-      - bundle-audit:
-          name: "ruby-2_6-bundle_audit"
-          e: "ruby_2_6"
-      - rubocop:
-          name: "ruby-2_6-rubocop"
-          e: "ruby_2_6"
-      - rspec-unit:
-          name: "ruby-2_6-rspec"
-          e: "ruby_2_6"
   ruby_2_7:
     jobs:
-      - bundle-audit:
-          name: "ruby-2_7-bundle_audit"
-          e: "ruby_2_7"
-      - rubocop:
-          name: "ruby-2_7-rubocop"
-          e: "ruby_2_7"
-      - rspec-unit:
-          name: "ruby-2_7-rspec"
-          e: "ruby_2_7"
+      - ruby/bundle-audit:
+          <<: *ruby_2_7_defaults
+          name: ruby-2_7-bundle_audit
+      - ruby/rubocop:
+          <<: *ruby_2_7_defaults
+          name: ruby-2_7-rubocop
+      - ruby/rspec-unit:
+          <<: *ruby_2_7_defaults
+          name: ruby-2_7-rspec_unit
+          db: false
+          code-climate: false
```

Comment:
❓ why are we running code climate against 3.0 but developing against 2.7?

Reply:
It doesn't really matter which version we run it against; it analyzes the code statically so it's irrespective of the Ruby version.

Since 2.7 EOL's next year, this is a bit of future proofing.